### ISSUE TITLE
Coredump on some sites under wayland

### ISSUE THREAD

**Issue by schyzophrene-asynchrone** [qutebrowser/qutebrowser#5313]
**Version info**:
```
qutebrowser v1.10.1
Git commit: 
Backend: QtWebEngine (Chromium 77.0.3865.129)
Qt: 5.14.1

CPython: 3.8.2
PyQt: 5.14.1

sip: 5.1.1
colorama: 0.4.3
pypeg2: 2.15
jinja2: 2.11.1
pygments: 2.6.1
yaml: 5.3
cssutils: no
attr: 19.3.0
PyQt5.QtWebEngineWidgets: yes
PyQt5.QtWebEngine: 5.14.0
PyQt5.QtWebKitWidgets: yes
pdf.js: no
sqlite: 3.31.1
QtNetwork SSL: OpenSSL 1.1.1e  17 Mar 2020

Style: QFusionStyle
Platform: Linux-5.5.13-arch1-1-x86_64-with-glibc2.2.5, 64bit
Linux distribution: Arch Linux (arch)
Frozen: False
Imported from /usr/lib/python3.8/site-packages/qutebrowser
Using Python from /usr/bin/python3
Qt library executable path: /usr/lib/qt/libexec, data path: /usr/share/qt

Paths:
cache: /home/hoel/testdir/cache
config: /home/hoel/testdir/config
data: /home/hoel/testdir/data
runtime: /home/hoel/testdir/runtime
system data: /usr/share/qutebrowser

Autoconfig loaded: yes
Config.py: no config.py was loaded
Uptime: 0:00:00
```
**Does the bug happen if you start with `--temp-basedir`?**:
IDK, however I did create a new basedir with `-B testdir` and it still happens (`testdir` did not exist before).
**Description**
On some specific websites, after loading a little, qutebrowser closes encountering a coredump. One can find a clean basedir [here](https://github.com/qutebrowser/qutebrowser/files/4387767/testdir.tar.gz) just after the crash.
I also have a stacktrace (using another clean basedir). The debug informations (before gdb starts) are:
```
21:39:31 DEBUG    commands   command:run:541 command called: open ['unity.com']
21:39:31 DEBUG    commands   command:run:555 Calling qutebrowser.browser.commands.CommandDispatcher.openurl(<qutebrowser.browser.commands.CommandDispatcher>, 'unity.com', False, False, False, False, None, False, False)
21:39:31 DEBUG    url        urlutils:get_path_if_valid:387 Checking if 'unity.com' is a path
21:39:31 DEBUG    url        urlutils:is_url:261 Checking if 'unity.com' is a URL (autosearch=naive).
21:39:31 DEBUG    url        urlutils:is_url:299 Checking via naive check
21:39:31 DEBUG    url        urlutils:is_url:303 url = True
21:39:31 DEBUG    url        urlutils:fuzzy_url:215 URL is a fuzzy address
21:39:31 DEBUG    url        urlutils:fuzzy_url:217 Converting fuzzy term 'unity.com' to URL -> http://unity.com
21:39:31 DEBUG    webview    browsertab:_on_before_load_started:961 Going to start loading: http://unity.com
21:39:31 DEBUG    webview    tabbedbrowser:_on_title_changed:750 Changing title for idx 0 to 'http://unity.com'
21:39:31 DEBUG    webview    browsertab:_on_before_load_started:961 Going to start loading: http://unity.com
21:39:31 DEBUG    webview    tabbedbrowser:_on_title_changed:750 Changing title for idx 0 to 'http://unity.com'
21:39:31 DEBUG    misc       eventfilter:eventFilter:55 <qutebrowser.browser.webengine.webview.WebEngineView object at 0x7fffd4148670> got new child <PyQt5.QtWidgets.QWidget object at 0x7fffd41584c0>, installing filter
21:39:31 DEBUG    modes      modeman:_handle_keypress:268 match: 2, forward_unbound_keys: auto, passthrough: True, is_non_alnum: True, dry_run: False --> filter: True (focused: <PyQt5.QtWidgets.QWidget object at 0x7fffd4148ee0>)
21:39:31 DEBUG    signals    signalfilter:_filter_signals:87 emitting: cur_url_changed(PyQt5.QtCore.QUrl('http://unity.com/')) (tab 0)
21:39:31 DEBUG    webview    browsertab:_set_load_status:927 load status for <qutebrowser.browser.webengine.webenginetab.WebEngineTab tab_id=0 url='http://unity.com/'>: LoadStatus.loading
21:39:31 DEBUG    signals    signalfilter:_filter_signals:87 emitting: cur_load_status_changed(<LoadStatus.loading: 6>) (tab 0)
21:39:31 DEBUG    signals    signalfilter:_filter_signals:87 emitting: cur_load_started() (tab 0)
21:39:31 DEBUG    webview    browsertab:_on_navigation_request:986 navigation request: url http://unity.com/, type Type.typed, is_main_frame True
21:39:31 DEBUG    webview    browsertab:_on_navigation_request:986 navigation request: url https://unity.com/, type Type.typed, is_main_frame True
21:39:31 DEBUG    misc       eventfilter:eventFilter:80 <qutebrowser.browser.webengine.webview.WebEngineView object at 0x7fffd4148670>: removed child <PyQt5.QtCore.QObject object at 0x7fffd4158820>
21:39:31 DEBUG    misc       eventfilter:eventFilter:55 <qutebrowser.browser.webengine.webview.WebEngineView object at 0x7fffd4148670> got new child <PyQt5.QtWidgets.QWidget object at 0x7fffd4158820>, installing filter
21:39:31 DEBUG    modes      modeman:_handle_keypress:245 got keypress in mode KeyMode.normal - delegating to <qutebrowser.keyinput.modeparsers.NormalKeyParser>
21:39:31 DEBUG    keyboard   basekeyparser:_debug_log:196 Got key: 0x1000001 / modifiers: 0x0 / text: '<Tab>' / dry_run False
21:39:31 DEBUG    keyboard   basekeyparser:_debug_log:196 Trying match without modifiers
21:39:31 DEBUG    keyboard   basekeyparser:_debug_log:196 Trying match with key_mappings
21:39:31 DEBUG    keyboard   basekeyparser:_debug_log:196 Giving up with '<Tab>', no matches
21:39:31 DEBUG    keyboard   basekeyparser:_debug_log:196 Clearing keystring (was: <Tab>).
21:39:31 DEBUG    modes      modeman:_handle_keypress:268 match: 0, forward_unbound_keys: auto, passthrough: False, is_non_alnum: True, dry_run: False --> filter: False (focused: <PyQt5.QtWidgets.QWidget object at 0x7fffd4148ee0>)
21:39:31 DEBUG    modes      modeman:_handle_keyrelease:293 filter: False
21:39:31 DEBUG    misc       app:on_focus_object_changed:488 Focus object changed: <PyQt5.QtWidgets.QWidget object at 0x7fffd4148f70>
21:39:31 DEBUG    misc       app:on_focus_object_changed:488 Focus object changed: <PyQt5.QtWidgets.QWidget object at 0x7fffd41584c0>
21:39:31 DEBUG    webview    tabbedbrowser:_on_title_changed:750 Changing title for idx 0 to 'https://unity.com'
21:39:31 DEBUG    signals    signalfilter:_filter_signals:87 emitting: cur_url_changed(PyQt5.QtCore.QUrl('https://unity.com/')) (tab 0)
21:39:31 DEBUG    modes      modeman:_handle_keyrelease:293 filter: True
21:39:31 DEBUG    misc       eventfilter:eventFilter:80 <qutebrowser.browser.webengine.webview.WebEngineView object at 0x7fffd4148670>: removed child <PyQt5.QtCore.QObject object at 0x7fffd4148f70>
[New Thread 0x7fff75dfc700 (LWP 8479)]
21:39:31 DEBUG    webview    tabbedbrowser:_on_title_changed:750 Changing title for idx 0 to 'Unity Real-Time Development Platform | 3D, 2D VR & AR Visualizations'

Thread 16 "Chrome_InProcGp" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fff9bfff700 (LWP 8439)]
0x0000000000000000 in ?? ()
```
I pasted it starting from the moment I entered `:open unity.com` on the command bar.
I also join the generated 
[gdb.txt](https://github.com/qutebrowser/qutebrowser/files/4389843/gdb.txt) file.

For additional information, I’m using wayland (with sway) and though xwayland is enabled I default to run everything under native wayland when available. Hence I have `QT_QPA_PLATFORM=wayland-egl`. Moreover unsetting it before starting qutebrowser causes the bug not to happen.
**How to reproduce**
Go to unity.com (for example, I can get more if needed, it happens quite a lot).

**Comment by bitraid** [comment]
Falkon is the same (crashes only under wayland). Seems that this might be a QT-wayland bug.

**Comment by pltrz** [comment]
this is also happening with me, under sway with xwayland disabled.

**Comment by The-Compiler** [comment]
Could you please:

- Run after `export WAYLAND_DEBUG=client` and show the log? (might need to `gzip` it)
- Add `disable-viz-display-compositor` to the `qt.args` setting and see if that changes anything?
- Set `content.webgl` to false and see if that does?

I haven't gotten around to trying to reproduce yet, so I'll probably try by myself later this week if nobody else does.

**Comment by bitraid** [comment]
- [Log](https://github.com/qutebrowser/qutebrowser/files/4410174/qb.log.gz)
- Still crashes with adding `disable-viz-display-compositor` to `qt.args`.
- No crash with `content.webgl` set to `False`
- No crash with javascript disabled.

Because the crash is similar to #5302, i also tried what's reported as a fix on QTBUG (disable the first `if` block of `QWaylandWindow::reset` in `qwaylandwindow.cpp` -  i didn't compile it though, i just patched `libQt5WaylandClient.so`), but the crash still happens for both issues.

**Comment by rosskarchner** [comment]
I don't know if this is related or not. I'm on Manjaro-arm(Sway), and scrolling on some sites (specifically nytimes.com) gets weird if I have qt5-wayland installed. If I'm using the J key to scroll down, sometimes it scrolls back up, and it's like I can never get to the bottom of the page.

If I uninstall that, everything works fine.

**Comment by The-Compiler** [comment]
@rosskarchner Probably not related to this issue (which is about a crash). I'm guessing Qt 5.14.2 will land in Manjaro soon - can you retry with that and open a new issue if it still happens there?

**Comment by The-Compiler** [comment]
I just tried to reproduce this with Qt 5.14.2 and I can't. @schyzophrene-asynchrone @bitraid @pltrz can you?

**Comment by pltrz** [comment]
Continues to happen with me on Qt 5.14.2 - only on an older machine with an i3-2330M. No problems on my more modern machine.

**Comment by The-Compiler** [comment]
If someone on Archlinux can still reproduce this, could you please get a gdb stacktrace again, but with debug symbols according to [this page](https://github.com/qutebrowser/qutebrowser/blob/master/doc/stacktrace.asciidoc) (with at least `qt5-{base,webengine,wayland}-debug` installed)? My debug packages are now up-to-date again.

**Comment by bitraid** [comment]
I can also reproduce on older IntelHD graphics only.
Log with symbols:
```
5:53:30 DEBUG    webview    tabbedbrowser:_on_title_changed:750 Changing title for idx 0 to 'Unity Real-Time Development Platform | 3D, 2D VR & AR Visualizations'

Thread 16 "Chrome_InProcGp" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffb4ff9700 (LWP 1266)]
0
```
[gdb.txt.gz](https://github.com/qutebrowser/qutebrowser/files/4481302/gdb.txt.gz)

**Comment by The-Compiler** [comment]
Awesome, thanks! I reported it upstream here: [[QTBUG-83493] Segfault in gpu::gles2::GLES2DecoderImpl::DoFramebufferParameteri when visiting unity.com on Wayland - Qt Bug Tracker](https://bugreports.qt.io/browse/QTBUG-83493)

**Comment by The-Compiler** [comment]
Is this reproducible when doing `qutebrowser --temp-basedir --qt-flag webgl-antialiasing-mode=none unity.com` as well?

**Comment by bitraid** [comment]
> Is this reproducible when doing `qutebrowser --temp-basedir --qt-flag webgl-antialiasing-mode=none unity.com` as well?

Yes.

**Comment by The-Compiler** [comment]
Reopening this as I think it'd be good to force-disable WebGL on affected Qt versions as a workaround. Has anyone had a chance to test this with Qt 5.15 (released today) yet?

**Comment by The-Compiler** [comment]
@schyzophrene-asynchrone @bitraid @pltrz @sentriz could you please show the output of running the following script on the machine/environment where you're getting this crash?

```python3
from PyQt5.QtGui import QOpenGLContext, QOpenGLVersionProfile, QOffscreenSurface, QGuiApplication

app = QGuiApplication([])

surface = QOffscreenSurface()
surface.create()

ctx = QOpenGLContext()
ok = ctx.create()
assert ok

ok = ctx.makeCurrent(surface)
assert ok

print("GLES: {}".format(ctx.isOpenGLES()))

vp = QOpenGLVersionProfile()
vp.setVersion(2, 0)

vf = ctx.versionFunctions(vp)
print("Vendor: {}".format(vf.glGetString(vf.GL_VENDOR)))
print("Renderer: {}".format(vf.glGetString(vf.GL_RENDERER)))
print("Version: {}".format(vf.glGetString(vf.GL_VERSION)))
print("Shading language version: {}".format(vf.glGetString(vf.GL_SHADING_LANGUAGE_VERSION)))

ctx.doneCurrent()
```

**Comment by bitraid** [comment]
```
GLES: False
Vendor: Intel Open Source Technology Center
Renderer: Mesa DRI Intel(R) HD Graphics 4000 (IVB GT2)
Version: 3.0 Mesa 20.0.7
Shading language version: 1.30
```

**Comment by sentriz** [comment]
```
GLES: False
Vendor: Intel Open Source Technology Center
Renderer: Mesa DRI Intel(R) HD Graphics 3000 (SNB GT2)
Version: 3.0 Mesa 20.0.6
Shading language version: 1.30
```

**Comment by pltrz** [comment]
sorry, I currently don't have access to this machine

**Comment by The-Compiler** [comment]
I now added a dialog which suggests to disable WebGL if on Wayland with older graphics hardware. If someone could confirm this isn't a problem anymore on Qt 5.15 (with WebGL enabled) that'd be great!

**Comment by bitraid** [comment]
No crash here with QT 5.15.

**Comment by The-Compiler** [comment]
Awesome, thanks for all the testing!

**Comment by schyzophrene-asynchrone** [comment]
```
GLES: False
Vendor: Intel Open Source Technology Center
Renderer: Mesa DRI Intel(R) HD Graphics 4000 (IVB GT2)
Version: 3.0 Mesa 20.0.7
Shading language version: 1.30```

**Comment by petabyteboy** [comment]
Is there still a way to keep WebGL enabled? I used qutebrowser with Qt 5.14.2 for a while now, also used WebGL sometimes and didn't notice any problems. I would like to keep using it on my own risk.

**Comment by The-Compiler** [comment]
@petabyteboy Oh, I wasn't aware that there are people with that configuration but without any issues. That's interesting... You should be able to set `QUTE_SKIP_WAYLAND_WEBGL_CHECK=1` as environment variable to ignore the check.

**Comment by petabyteboy** [comment]
This environment variable allows me to enable WebGL again, and I was able to run all tested WebGL workloads without any issues.
